temba/channels/handlers.py

```diff
@@ -1954,3 +1954,42 @@ def post(self, request, *args, **kwargs):
 
         else:  # pragma: no cover
             return HttpResponse("Not handled, unknown action", status=400)
+
+
+class LineHandler(View):
+
+    @disable_middleware
+    def dispatch(self, *args, **kwargs):
+        return super(LineHandler, self).dispatch(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        return self.post(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        from temba.msgs.models import Msg
+
+        channel_uuid = kwargs['uuid']
+
+        channel = Channel.objects.filter(uuid=channel_uuid, is_active=True, channel_type=Channel.TYPE_LINE).exclude(org=None).first()
+        if not channel:
+            return HttpResponse("Channel with uuid: %s not found." % channel_uuid, status=404)
+
+        try:
+            data = request.body.decode('utf-8', errors='ignore')
+            content = json.loads(data)
+            events = content.get('events')
+
+            for item in events:
+                source = item.get('source')
+                message = item.get('message')
+
+                if message.get('type') == 'text':
```

Comment:
Should validate and fail with 400 if source, message or type are not in the request.